include/zephyr/toolchain/xcc.h

```diff
@@ -18,6 +18,21 @@
 #define __BYTE_ORDER__
 #endif
 
+#ifdef __clang__
+#if __clang_major__ >= 10
+#define __fallthrough __attribute__((fallthrough))
+#endif
```

Comment:
Don't you need an #else clause here so that `__fallthrough` is never undefined?

Reply:
Should be okay since `include/zephyr/toolchain/gcc.h` (included below) will define `__fallthrough` if it is not already done so.